rptest: add `test_add_and_decommission()`

andrewhsu opened the pull request:
Fixes https://github.com/redpanda-data/cloudv2/issues/10807

Blocked by PR: https://github.com/redpanda-data/vtools/pull/2407

Example command to run new test:
```console
ducktape \
  --debug \
  --globals=/home/ubuntu/redpanda/tests/globals.json \
  --cluster=ducktape.cluster.json.JsonCluster \
  --cluster-file=/home/ubuntu/redpanda/tests/cluster.json \
  --test-runner-timeout=3600000 \
  tests/rptest/redpanda_cloud_tests/high_throughput_test.py::HighThroughputTest.test_add_and_decommission
```

output from test run:
```
test_id:    rptest.redpanda_cloud_tests.high_throughput_test.HighThroughputTest.test_add_and_decommission
status:     PASS
run time:   11 minutes 35.892 seconds
--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
============================================================================================================================================================================================
SESSION REPORT (ALL TESTS)
ducktape version: 0.8.18
session_id:       2023-12-21--006
run time:         11 minutes 35.913 seconds
tests run:        1
passed:           1
flaky:            0
failed:           0
ignored:          0
opassed:          0
ofailed:          0
==========================================================================================================================================================================================
```

## Backports Required

<!-- Checking at least one of the checkboxes is REQUIRED if this PR is not a backport. -->

- [x] none - not a bug fix
- [ ] none - this is a backport
- [ ] none - issue does not exist in previous branches
- [ ] none - papercut/not impactful enough to backport
- [ ] v23.3.x
- [ ] v23.2.x
- [ ] v23.1.x

## Release Notes

* none

andrewhsu commented:
test run output coming soon (need to rebuild dev env), but in the meantime this PR is ready for review

savex reviewed (approved):
lgtm on green build.

side note: I would move in the future the tests that related to day-2 operations to separate module to tidy up. Something like `redpanda_cloud_tests/htt_operations_tests.py` and rename this file to `htt_limits_test.py`. That would be more meaningful in scope what tests are actually doing.

vbotbuildovich commented:
<!--NEW FAILURES-->
new failures in https://buildkite.com/redpanda/redpanda/builds/43054#018c82d7-0744-40ec-9576-7f4bfa152f9c:
```
"rptest.tests.cloud_storage_usage_test.CloudStorageUsageTest.test_cloud_storage_usage_reporting"
```
<!--e98994f6-5daa-479c-acad-e7606dc191e5-->

vbotbuildovich commented:
<!--RETRIED FAILURES-->
ducktape was retried in https://buildkite.com/redpanda/redpanda/builds/43077#018c844d-2794-47c8-8603-48d611186154
<!--289ceb67-5b1b-4949-94f2-538cb1301a4a-->

andrewhsu commented:
needs adjustment on the systemctl services. putting this back into draft mode for now while i'll get an update for that.

andrewhsu commented:
fixed the systemctl service. the agent services have to be running when making the call to admin-api to scale nodes.

i've also added a successful test output to the PR description. ready for review.

savex commented on tests/rptest/redpanda_cloud_tests/high_throughput_test.py:
```diff
@@ -1056,6 +1063,191 @@ def deployment_ready_replicas():
+        self._delete_cluster_pvc(cluster_name, orig_replicas)
+        self._patch_deployment_args('--allow-downscaling=true',
+                                    '--allow-downscaling=false')
+        self._enable_agent_services()
```
this is a clean-good-looking test code!

savex commented on tests/rptest/redpanda_cloud_tests/high_throughput_test.py:
```diff
@@ -1056,6 +1063,191 @@ def deployment_ready_replicas():
+    def _patch_cluster_replicas(self, cluster_name, replicas):
+        def cluster_ready_replicas(cluster_name):
+            # kubectl get cluster rp-clkd0n22nfn1jf7vd9t0 -n=redpanda -o=jsonpath='{.status.readyReplicas}'
+            return int(
```
what is the difference here with line 1119?

savex reviewed (commented):
looks good. One doubled code though
